### ISSUE TITLE
abacus-dataflow returns 201 even if SINK is unsuccessful

### ISSUE THREAD

**Issue by rajkiranrbala** [cloudfoundry-incubator/cf-abacus#245]
I am using abacus-dataflow module to extend the pipleline beyond abacus-usage-aggregator. I noticed that the collector returns a status code of 201 even if the SINK app responds with a status code in the range 400 to 499 excluding 409.  

When I was looking at the dataflow module the [postOutput](https://github.com/cloudfoundry-incubator/cf-abacus/blob/master/lib/utils/dataflow/src/index.js#L302-L318)  I saw that ```yield brequest.post(...``` throws an exception when the response status code is >= 500. Otherwise the function returns without checking for status code.

```javascript
const postOutput = function *(olog, shost, spartition, spost, authentication) {
  if(!spost || !shost)
    return;
  const phost = yield sink(olog.id, shost, spartition);

  debug('Posting output doc %s to %s', olog.id, phost + spost);
  const res = yield brequest.post(phost + spost,
    extend(authentication ? {
      headers: {
        authorization: authentication()
      }
    } : {}, {
      body: olog
    }));
  debug('Posted %s to sink, response status %d', olog.id, res.statusCode);
};
```

I don't want to return a response with staus code 500. Can the dataflow module be modified to accept a function as an other parameter which is gonna process the response from the SINK  and return a value based on that?

**Comment by rajkiranrbala** [comment]
@jsdelfino Any thoughts on the implementation other than having a user supplied handler?

**Comment by jsdelfino** [comment]
The dataflow module should simply propagate all errors from the target sink. I'll work on a fix.

**Comment by rajkiranrbala** [comment]
Lets say the mapper/reducer produces three documents for a single document. If all the three documents are posted to the SYNC and they have 2 that failed, do we propagate both the error codes?

**Comment by jsdelfino** [comment]
Fixed by 74a78a0.
